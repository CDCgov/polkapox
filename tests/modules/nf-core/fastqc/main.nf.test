nextflow_process {

    name "FASTQC Test"
    script "../../../../modules/nf-core/fastqc/main.nf"
    process "FASTQC"

    tag "fastqc"

    test("Single-end test") {
        when {
            process {
                """
                input[0] = Channel.of([
                    [ id: 'test_sub', single_end: true ],
                    file('${projectDir}/tests/nf-test-data/test_sub_1.fq.gz', checkIfExists: true)
                ])
                """
            }
        }

        then {
            assert process.success

            if (process.success) {
                assert process.success
                assert snapshot(process.out).match()
            }
        }
    }

}
