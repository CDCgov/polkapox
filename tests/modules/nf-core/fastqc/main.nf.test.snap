{
    "Paired-end test": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "2": [
                    
                ],
                "html": [
                    
                ],
                "versions": [
                    
                ],
                "zip": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-07-09T14:59:16.343967081"
    },
    "Single-end test": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "2": [
                    
                ],
                "html": [
                    
                ],
                "versions": [
                    
                ],
                "zip": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-07-09T14:59:08.989471133"
    }
}